cmd/clusterctl/pkg/client/cluster/mover.go

```diff
@@ -84,6 +90,54 @@ func newObjectMover(fromProxy Proxy, log logr.Logger) *objectMover {
 	}
 }
 
+// checkProvisioningCompleted checks if Cluster API has already completed the provisioning of the infrastructure for the objects involved in the move operation.
+func (o *objectMover) checkProvisioningCompleted(graph *objectGraph) error {
+	errList := []error{}
+	cFrom, err := o.fromProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// Checking all the clusters have infrastructure is ready
+	for _, cluster := range graph.getClusters() {
+		clusterObj := &clusterv1.Cluster{}
+		clusterObjKey := client.ObjectKey{
+			Namespace: cluster.identity.Namespace,
+			Name:      cluster.identity.Name,
+		}
+
+		if err := cFrom.Get(ctx, clusterObjKey, clusterObj); err != nil {
+			return errors.Wrapf(err, "error reading %q %s/%s",
+				clusterObj.GroupVersionKind(), clusterObj.GetNamespace(), clusterObj.GetName())
+		}
+
+		if !clusterObj.Status.InfrastructureReady {
+			errList = append(errList, errors.Errorf("cannot start the move operation while  %q %s/%s is still provisioning the infrastructure", clusterObj.GroupVersionKind(), clusterObj.GetNamespace(), clusterObj.GetName()))
+		}
+	}
+
+	// Checking all the clusters have a NodeRef
+	// Nb. NodeRef is considered a better signal that InfrastructureReady, because it ensures the node in the workload cluster is up and running.
```

Comment:
nit:
```suggestion
	// Nb. NodeRef is considered a better signal than InfrastructureReady, because it ensures the node in the workload cluster is up and running.
```